### PR TITLE
switch CI to github actions

### PR DESCRIPTION
**Pull request by johnnychen94**
The contents are copied from OffsetArrays.jl

### CODE REVIEW

File: .github/workflows/UnitTest.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        julia-arch: [x64]
+        # only test one 32-bit job
+        include:
+          - os: ubuntu-latest
+            julia-version: '1'
+            julia-arch: x86
+
+    steps:
+      - uses: actions/checkout@v1.0.0
```
Comment (rikhuijzer) [inline .github/workflows/UnitTest.yml:37]: No reason that I know of to use the old version. See <https://github.com/marketplace/actions/checkout>:

```suggestion
      - uses: actions/checkout@v2
```

File: .github/workflows/UnitTest.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          cache-name: cache-artifacts
+        with:
+          path: ~/.julia/artifacts 
+          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+      - name: "Unit Test"
+        uses: julia-actions/julia-runtest@master
```
Comment (rikhuijzer) [inline .github/workflows/UnitTest.yml:56]: From https://github.com/marketplace/actions/run-julia-package-tests:

```suggestion
        uses: julia-actions/julia-runtest@v1
```

**Review by rikhuijzer — COMMENTED** [review]
I'm not an expert, but an extra set of eyes won't hurt, I thought.

**Review by rikhuijzer — APPROVED** [review]

**Comment by johnnychen94** [comment]
Sorry! I was caught by too many stuff last week

**Pull request merged**